gui/default/syncthing/core/syncthingController.js

```diff
@@ -912,6 +923,17 @@ angular.module('syncthing.core')
             return '';
         };
 
+        $scope.findDeviceFromShort = function (shortID) {
+            var matches = $scope.devices.filter(function (n) {
+                return n.deviceID.substr(0, 7) === shortID;
+            });
+            if (matches.length !== 1) {
+                n.deviceID.name = "Unknown";
+                return n;
+            }
+            return matches[0];
+        };
+
```

Comment:
I think this is what we're talking about, right, combined with just handling the empty string in the GUI:

```diff
diff --git a/lib/protocol/deviceid.go b/lib/protocol/deviceid.go
index cab70e3..e2b5d15 100644
--- a/lib/protocol/deviceid.go
+++ b/lib/protocol/deviceid.go
@@ -88,6 +88,9 @@ func (n *DeviceID) MarshalText() ([]byte, error) {
 }
 
 func (s ShortID) String() string {
+       if s == 0 {
+               return ""
+       }
        var bs [8]byte
        binary.BigEndian.PutUint64(bs[:], uint64(s))
        return base32.StdEncoding.EncodeToString(bs[:])[:7]
```